### PR TITLE
Recompute the device-limit view when the device is removed - VPN-2847

### PR DESCRIPTION
**Pull request by bakulf**
I moved the logic in `deviceRemoved()`

### CODE REVIEW

File: src/mozillavpn.cpp [hunk 1]
```diff
@@ -662,36 +681,19 @@ void MozillaVPN::removeDeviceFromPublicKey(const QString& publicKey) {
...
-    }
+  // Let's inform the UI about what is going to happen.
+  emit deviceRemoving(publicKey);
+  TaskScheduler::scheduleTask(new TaskRemoveDevice(publicKey));
 
-    maybeStateMain();
-  }));
+  if (m_state != StateDeviceLimit) {
+    // To have a faster UI, we inform the device-model that this public key
+    // is going to be removed.
```
Comment (brizental) [inline src/mozillavpn.cpp:694]: At which point is the UI informed about the device removal, here, on line 688 or both? And if both, why? This is confusing.

**Review by brizental — CHANGES_REQUESTED** [review]
Would it be possible to write a test for this behaviour?

File: src/inspector/inspectorhandler.cpp [hunk 2]
```diff
@@ -939,6 +940,22 @@ static QList<InspectorCommand> s_commands{
...
+          payload["publicKey"] = QString(arguments[1]);
+
+          QJsonObject msg;
+          msg["type"] = "DEVICE_DELETED";
+          msg["payload"] = payload;
+
+          PushMessage message(QJsonDocument(msg).toJson());
+          message.executeAction();
+          return QJsonObject();
+        }},
```
Comment (brizental) [inline src/inspector/inspectorhandler.cpp:958]: Nice!

**Review by brizental — APPROVED** [review]

File: tests/functional/testDevices.js [hunk 3]
```diff
@@ -169,8 +169,126 @@ describe('Devices', function() {
...
+      await vpn.waitForElementProperty('deviceListView', 'visible', 'true');
+
+      await vpn.waitForElement('deviceLimitHeader');
+      await vpn.waitForElementProperty('deviceLimitHeader', 'visible', 'true');
+
+      const key = UserData.devices[0].pubkey;
+      UserData.devices.splice(0, 1);
+      await vpn.sendPushMessageDeviceDeleted(key);
+
+      await vpn.waitForElement('controllerTitle');
```
[the remark this answers is not in the document]
Comment (bakulf) [reply tests/functional/testDevices.js:287]: It means that the UI shows the controller main view. The title is "VPN is On" or things like this.

**Pull request merged**